pennylane/transforms/dynamic_one_shot.py

```diff
@@ -288,6 +308,39 @@ def measurement_with_no_shots(measurement):
     return tuple(normalized_meas) if len(normalized_meas) > 1 else normalized_meas[0]
 
 
+def gather_mcm_qjit(measurement, samples, is_valid):  # pragma: no cover
+    """Process MCM measurements when the Catalyst compiler is active.
+
+    Args:
+        measurement (MeasurementProcess): measurement
+        samples (dict): Mid-circuit measurement samples
+        is_valid (TensorLike): Boolean array with the same shape as ``samples`` where the value at
+            each index specifies whether or not the respective sample is valid.
+
+    Returns:
+        TensorLike: The combined measurement outcome
+    """
+    found, meas = False, None
+    for k, meas in samples.items():
+        if measurement.mv is k.out_classical_tracers[0]:
+            found = True
+            break
+    if not found:
+        raise LookupError("MCM not found")
+    meas = qml.math.squeeze(meas)
+    if isinstance(measurement, (CountsMP, ProbabilityMP)):
+        sum_valid = qml.math.sum(is_valid)
+        count_1 = qml.math.sum(meas * is_valid)
+        if isinstance(measurement, CountsMP):
+            return {0: sum_valid - count_1, 1: count_1}
```

Comment:
I think this breaks the return type convention of counts in catalyst, which is not a dictionary. I'm not sure how serious this is though, but if in doubt would this be acceptable? 

```suggestion
            return qml.math.array([0, 1]), qml.math.array([sum_valid - count_1, count_1])
```

Or does it break further processing code?

Reply:
This was to sync with PennyLane as much as possible, but I can return arrays without issue. I'll just have to fix how the tests validate measurements.